### PR TITLE
hiop: update constraints

### PR DESCRIPTION
**Pull request by ashermancinelli**
Changes required to be compatible with newest umpire+cuda.

CC @CameronRutherford

### CODE REVIEW

**Review by cameronrutherford — DISMISSED** [review]
LGTM 👍🏻

File: var/spack/repos/builtin/packages/hiop/package.py [hunk 1]
```diff
@@ -68,27 +65,34 @@ class Hiop(CMakePackage, CudaPackage):
...
-        lapack_blas_libs = (
-            spec['lapack'].libs + spec['blas'].libs).joined(';')
-        args.extend([
-            '-DLAPACK_FOUND=TRUE',
-            '-DLAPACK_LIBRARIES={0}'.format(lapack_blas_libs)
-        ])
+        lapack_blas_libs = (spec['lapack'].libs + spec['blas'].libs).joined(';')
+        args.extend(
+            ['-DLAPACK_FOUND=TRUE', '-DLAPACK_LIBRARIES={0}'.format(lapack_blas_libs)]
+        )
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hiop/package.py]: Prefer using the cmake package helper functions, which convert bools and lists to their appropriate CMake string representation:
```suggestion
        args.extend([
            self.define('LAPACK_FOUND', True),
            self.define('LAPACK_LIBRARIES',
                        spec['lapack'].libs + spec['blas'].libs)
        ])
```

File: var/spack/repos/builtin/packages/hiop/package.py [hunk 2]
```diff
@@ -68,27 +65,34 @@ class Hiop(CMakePackage, CudaPackage):
...
 
+        args.append('-DHIOP_BUILD_STATIC=ON')
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hiop/package.py]: Same here. You also might as well combine all these `append`s below into the `extends` above.

**Comment by ashermancinelli** [comment]
@sethrj  Thanks for the recommendations! I've committed them and cleaned up the package a bit. I'll test out the install and push shortly.

**Comment by ashermancinelli** [comment]
@sethrj Thank you for those awesome suggestions! They make the package much more readable imo. I've applied them liberally.

File: var/spack/repos/builtin/packages/hiop/package.py [hunk 3]
```diff
@@ -7,11 +7,11 @@
 
 
 class Hiop(CMakePackage, CudaPackage):
-    '''HiOp is an optimization solver for solving certain mathematical
+    """HiOp is an optimization solver for solving certain mathematical
     optimization problems expressed as nonlinear programming problems.
-    HiOp is a lightweight HPC solver that leverages application's existing
+    HiOp is a lightweight HPC solver that leverages application"s existing
```
Comment (sethrj) [inline var/spack/repos/builtin/packages/hiop/package.py:12]: Find-replace gone awry. The double quotes are OK and in fact Spack and its packages will be reformatted sometime soonish using black, at which point the'll all become double quotes.

**Review by sethrj — APPROVED** [review]
Looks great to me, aside from the find-replace issue ;)

**Pull request merged**